roles/preflight-checks/tasks/subnet.yml

```diff
@@ -0,0 +1,63 @@
+---
+- name: checking neutron subnet {{ item.name }}
+  os_subnets_facts:
+    name: "{{ item.name }}"
+    auth:
+      auth_url: "{{ endpoints.auth_uri }}"
+      project_name: admin
+      username: admin
+      password: "{{ secrets.admin_password }}"
+
+- name: ensure subnet name
+  assert: 
+    that: openstack_subnets[0] is defined
+    msg: "The subnet does not exist or name may have been changed."
+
+- name: ensure cidr
+  assert: 
+    that: openstack_subnets[0]['cidr'] == item.cidr
+    msg: "Could not find matching attribute for cidr."
+
+- name: ensure gateway_ip
+  assert:
+    that: openstack_subnets[0]['gateway_ip'] == item.gateway_ip
+    msg: "Could not find matching attribute for gateway_ip."
+
+- name: ensure ip_version
+  assert:
+    that: openstack_subnets[0]['ip_version'] == item.ip_version
+    msg: "Could not find matching attribute for ip_version."
+
+- name: ensure enable_dhcp
+  assert:
+    that: openstack_subnets[0]['enable_dhcp'] == {{ item.enable_dhcp | bool }}
+    msg: "Could not find matching attribute for enable_dhcp."
+
+- name: ensure ipv6_ra_mode
+  assert:
+    that: openstack_subnets[0]['ipv6_ra_mode'] == item.ipv6_ra_mode
+    msg: "Could not find matching attribute for ipv6_ra_mode."
+  when: item.ipv6_ra_mode is defined
+
+- name: ensure pool_start
+  assert:
+    that: openstack_subnets[0]['allocation_pools'][0]['start'] == item.pool_start
+    msg: "Could not find matching attribute for pool_start."
+  when: item.pool_start is defined
+
+- name: ensure pool_end
+  assert:
+    that: openstack_subnets[0]['allocation_pools'][0]['end'] == item.pool_end
+    msg: "Could not find matching attribute for pool_end."
+  when: item.pool_end is defined
+
+- name: ensure ipv6_address_mode
+  assert:
+    that: openstack_subnets[0]['ipv6_address_mode'] == item.ipv6_address_mode
+    msg: "Could not find matching attribute for ipv6_address_mode."
+  when: item.ipv6_ra_mode is defined
+
+#FIXME: Add dns_nameservers check, there could be zero to N number of DNS nameservers.
+#       In most cases, there are two or three.
+#       openstack_subnets[0]['dns_nameservers'][0] in item.dns_nameservers
+#       openstack_subnets[0]['dns_nameservers'][1] in item.dns_nameservers
```

Comment:
What's preventing this from being fixed? Adding that to the comment would be helpful for the next person looking at it.